src/SFA.DAS.AAN.Hub.Jobs/Services/SynchroniseApprenticeDetailsService.cs

```diff
@@ -0,0 +1,154 @@
+﻿using Microsoft.Extensions.Logging;
+using RestEase;
+using SFA.DAS.AAN.Hub.Data.Entities;
+using SFA.DAS.AAN.Hub.Data.Interfaces;
+using SFA.DAS.AAN.Hub.Jobs.Api.Clients;
+using SFA.DAS.AAN.Hub.Jobs.Api.Response;
+using SFA.DAS.AAN.Hub.Jobs.Functions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AAN.Hub.Jobs.Services;
+
+public interface ISynchroniseApprenticeDetailsService
+{
+    Task<int> SynchroniseApprentices(CancellationToken cancellationToken);
+}
+
+public class SynchroniseApprenticeDetailsService : ISynchroniseApprenticeDetailsService
+{
+    private readonly ILogger<SynchroniseApprenticeDetailsService> _logger;
+    private readonly IApprenticeAccountsApiClient _apprenticeAccountsApiClient;
+    private readonly IApprenticeRepository _apprenticeshipRespository;
+    private readonly IJobAuditRepository _jobAuditRepository;
+    private readonly IMemberRepository _memberRepository;
+
+    public SynchroniseApprenticeDetailsService(
+        ILogger<SynchroniseApprenticeDetailsService> logger,
+        IApprenticeAccountsApiClient apprenticeAccountsApiClient,
+        IApprenticeRepository apprenticeshipRespository,
+        IJobAuditRepository jobAuditRepository,
+        IMemberRepository memberRepository
+    )
+    {
+        _logger = logger;
+        _apprenticeAccountsApiClient = apprenticeAccountsApiClient;
+        _apprenticeshipRespository = apprenticeshipRespository;
+        _jobAuditRepository = jobAuditRepository;
+        _memberRepository = memberRepository;
+    }
+
+    public async Task<int> SynchroniseApprentices(CancellationToken cancellationToken)
+    {
+        JobAudit audit = new JobAudit()
+        {
+            JobName = nameof(SynchroniseApprenticeDetailsFunction),
+            StartTime = DateTime.UtcNow
+        };
+
+        try
+        {
+            var apprentices = await _apprenticeshipRespository.GetApprentices(cancellationToken);
+
+            if (apprentices is null || apprentices.Count == 0)
+            {
+                await RecordAudit(audit, null, cancellationToken);
+                return default;
+            }
+
+            var response = await QueryApprenticeApi(apprentices, cancellationToken);
+
+            if(response is null)
```

Comment:
Response will never be null, you can check http status code is not success, but in this case we do not want to create audit record